frontend/javascripts/oxalis/model/actions/dataset_actions.ts

```diff
@@ -5,12 +5,14 @@ type SetLayerTransformsAction = ReturnType<typeof setLayerTransformsAction>;
 export type EnsureLayerMappingsAreLoadedAction = ReturnType<
   typeof ensureLayerMappingsAreLoadedAction
 >;
+type SetLayerHasSegmentIndexAction = ReturnType<typeof setLayerHasSegmentIndexAction>;
 
 export type DatasetAction =
   | SetDatasetAction
   | SetLayerMappingsAction
   | SetLayerTransformsAction
-  | EnsureLayerMappingsAreLoadedAction;
+  | EnsureLayerMappingsAreLoadedAction
+  | SetLayerHasSegmentIndexAction;
```

Comment:
`EnsureSegmentIndexIsLoadedAction` is missing here (also needs to be defined probably).

Reply:
actually I did this on purpose because this action is not used in a reducer and so I thought it wouldn't be necessary to include this in the union types. will add this so that the action types remain complete